src/onepiece_classify/infer/recognition.py

```diff
@@ -1,28 +1,80 @@
+from PIL import Image
+import numpy as np
+import torch
+from pathlib import Path
+from .base import BaseInference
+from typing import Optional, Tuple, Dict
 
+from onepiece_classify.models import ImageRecogModel
+from onepiece_classify.data import OnepieceImageDataLoader
+from onepiece_classify.transforms import get_test_transforms
 
 
-
-from .base import BaseInference
-import torch
-
 class ImageRecognition(BaseInference):
     
-    def __init__(self, model_path: str):
-        self.model_path = model_path
-        
-        self.model = self._build_model(model_path)
+    def __init__(self, model_path: str, data_path: str):
+        self.model_path = Path(model_path)
+        self.data_path = Path(data_path)
+
+        self.class_dict = {
+            i: val for val, i in OnepieceImageDataLoader(self.data_path).trainset.class_to_idx.items()
+        }
+        self.name_class = OnepieceImageDataLoader(self.data_path).trainset.classes
+        self.nclass = len(self.name_class)
+        # self.model = self._build_model()
         
-    def _build_model(self, model_path):
+    def _build_model(self):
          # load model
-        state_dict = torch.load(model_path)
-
-        model = ImageModel.load(state_dict)
-        return model
+        state_dict = torch.load(self.model_path)
+        model_backbone = ImageRecogModel(self.nclass).build_backbone()
+        model_backbone.load_state_dict(state_dict)
+        return model_backbone
     
+    def pre_process(self, image: Optional[str | np.ndarray | Image.Image]) -> torch.Tensor:
+        
+        trans = get_test_transforms()
+
+        if isinstance(image, str):
+            img = Image.open(image).convert("RGB")
+            img = trans(img).unsqueeze(0)
+        
+        elif isinstance(image, Image.Image):
+            img = image.convert("RGB")
+            img = trans(img).unsqueeze(0)
+
+        elif isinstance(image, np.ndarray):
+            img = image.astype(np.uint8)
+            img = Image.fromarray(img).convert("RGB")
+            img = trans(img).unsqueeze(0)
+
+        else:
+            print("Image type not recognized")
+
+        return img
+
     def forward(self, image_tensor: torch.Tensor) -> torch.Tensor:
-        return self.model.forward(image_tensor)
-    
+        model = self._build_model()
```

Comment:
this is very wrong, it computational expensive, with the you write the code it will always load the model everytime it doing forward computation. My suggestion is load the model in the constructor instead.